pkg/instances/multi_ig_pool.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instances
+
+import (
+	compute "google.golang.org/api/compute/v1"
+	"k8s.io/ingress-gce/pkg/utils"
+	"k8s.io/ingress-gce/pkg/utils/namer"
+	"k8s.io/klog"
+)
+
+type MultiIGNodePool struct {
+	*Instances
+}
+
+func NewMultiIGNodePool(cloud InstanceGroups, namer namer.BackendNamer, recorders recorderSource, basePath string, zl ZoneLister) *MultiIGNodePool {
+	mignp := &MultiIGNodePool{
+		Instances: &Instances{
+			cloud:              cloud,
+			namer:              namer,
+			recorder:           recorders.Recorder(""), // No namespace
+			instanceLinkFormat: basePath + "zones/%s/instances/%s",
+			ZoneLister:         zl,
+		},
+	}
+	return mignp
+}
+
+func (igc *MultiIGNodePool) DeleteInstanceGroup(name string) error {
+	klog.Infof("DeleteInstanceGroup is a no-op. Instance groups will be deleted when the cluster is deleted.")
+	return nil
+}
+
+func (igc *MultiIGNodePool) Sync(nodeNames []string) error {
+	klog.Infof("Sync is a no-op. Instance groups will be synced in the separate controller.")
+	return nil
+}
+
+func (igc *MultiIGNodePool) Get(name, zone string) ([]*compute.InstanceGroup, error) {
+	var igs []*compute.InstanceGroup
+	igsForZone, err := igc.cloud.ListInstanceGroups(zone)
+	if err != nil {
+		return nil, err
+	}
+	for _, ig := range igsForZone {
+		if igc.namer.NameBelongsToCluster(ig.Name) {
+			igs = append(igs, ig)
+		}
+	}
+	return igs, nil
+}
+
+func (igc *MultiIGNodePool) List() ([]string, error) {
+	return igc.Instances.List()
+}
```

Comment:
Maybe we can remove this, if it just uses `Instances.List()` ?

Reply:
of course!